api_test.go

```diff
@@ -1749,3 +1751,126 @@ func TestCreateSiteRulesRateLimitClientIdentifiers(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestCRUDSiteRequestRule(t *testing.T) {
+	createSiteRulesBody := CreateSiteRuleBody{
+		Type:           "request",
+		GroupOperator:  "all",
+		Enabled:        true,
+		Reason:         "Example site rule",
+		RequestLogging: "none",
+		Expiration:     "",
+		Conditions: []Condition{
+			{
+				Type:     "single",
+				Field:    "ip",
+				Operator: "equals",
+				Value:    "1.2.3.4",
+			},
+			{
+				Type:          "group",
+				GroupOperator: "any",
+				Conditions: []Condition{
+					{
+						Type:     "single",
+						Field:    "ip",
+						Operator: "equals",
+						Value:    "5.6.7.8",
+					},
+				},
+			},
+		},
+		Actions: []Action{
+			{
+				Type: "block",
+			},
+		},
+	}
+
+	sc := NewTokenClient(testcreds.email, testcreds.token)
+	corp := testcreds.corp
+	site := testcreds.site
+
+	createResp, err := sc.CreateSiteRule(corp, site, createSiteRulesBody)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !compareSiteRuleBody(createSiteRulesBody, createResp.CreateSiteRuleBody) {
+		t.Errorf("CreateSiteRules got:\n %#v\n want\n %#v", createResp, createSiteRulesBody)
+	}
+
+	readResp, err := sc.GetSiteRuleByID(corp, site, createResp.ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !compareSiteRuleBody(createSiteRulesBody, readResp.CreateSiteRuleBody) {
+		t.Errorf("CreateSiteRules got:\n %#v\n want\n %#v", createResp, createSiteRulesBody)
+	}
+
+	updateSiteRuleBody := CreateSiteRuleBody{
+		Type:           "request",
+		GroupOperator:  "all",
+		Enabled:        true,
+		Reason:         "Example site rule",
+		Expiration:     "",
+		RequestLogging: "none",
+		Conditions: []Condition{
+			{
+				Type:     "single",
+				Field:    "ip",
+				Operator: "equals",
+				Value:    "1.2.3.4",
+			},
+			{
+				Type:          "group",
+				GroupOperator: "any",
+				Conditions: []Condition{
+					{
+						Type:     "single",
+						Field:    "ip",
+						Operator: "equals",
+						Value:    "9.10.11.12",
+					},
+				},
+			},
+		},
+		Actions: []Action{
+			{
+				Type: "block",
+			},
+		},
+	}
```

Comment:
It took a minute for me to realize what is changing for this test request body. To make this clearer, here's a suggestion thats more specific. We just want to know that this SDK is sending and retrieving rules correctly.

```suggestion
	updateSiteRuleBody := createSiteRulesBody
	updateSiteRuleBody.Reason = "a new reason"
```